build-release-tools/lib/manifest.py

```diff
@@ -45,7 +45,16 @@ def __init__(self, file_path, git_credentials = None):
         self.parse_manifest()
 
     @staticmethod
-    def instance_of_sample(manifest_sample="manifest.json"):
+    def instance_of_sample(type="rackhd"):
+        if type == "rackhd":
+            manifest_sample = "manifest.json"
+        elif type == "pr-gate":
+            manifest_sample = "manifest-pr-gate.json"
+        elif type == "on-web-ui":
+            manifest_sample = "manifest-web-ui.json"
+        else:
+            raise KeyError("No manifest template for type: {0}".format(type))
+
         repo_dir = os.path.dirname(sys.path[0])
```

Comment:
So this is a experimental PR.
**You can regard such lib changes of this PR as the mark of refactor.**
When we have a pipeline framework, these key-value may be just stored in the config file.
No code changes will be made.